### PR TITLE
Check webpack build on test

### PR DESCRIPTION
**Pull request by leggsimon**
A couple of releases happened that failed once merged because it only failed during the `webpack --bail --config build/deploy/webpack.config.js` command which lived within `make deploy`.
`make deploy` only happened on master and therefore it didn't fail on branches. This should ensure we catch these errors on branch builds

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -92,7 +95,7 @@ endif
 test-dev: verify test-browser-dev test-webpack
 
 deploy:
-	webpack --bail --config build/deploy/webpack.config.js
+	make build-webpack
```
Comment (adambraimbridge) [inline Makefile]: Should this be `$(MAKE) build-webpack`?

Comment (bjfletcher) [inline Makefile]: `deploy: build-webpack` is more Make-style

File: Makefile [hunk 2]
```diff
@@ -77,7 +80,7 @@ a11y: test-build pally-conf
 
 # Note: `run` executes `node demo/app`, which fires up express, then deploys
 # a test static site to s3, then exits, freeing the process to execute `nightwatch a11y`.
-test: developer-note verify pally-conf test-server test-browser test-build test-webpack run nightwatch a11y
+test: developer-note verify pally-conf test-server test-browser test-build test-webpack run nightwatch a11y build-webpack
```
Comment (bjfletcher) [inline Makefile]: The CircleCI app preset says to run during test: `make build-production test provision`

I wonder if it'd make better sense to do this instead:

```
build-production: build-webpack
```

and remove the `webpack --bail --config build/deploy/webpack.config.js` build line in the `deploy` target.

So it's in keeping with how we do it elsewhere in Next.

Disclaimer: I've barely worked with `n-ui` and if `n-ui` needs to be different then cool 👍

File: Makefile (hunk 1 above)
Comment (leggsimon) [inline Makefile]: @bjfletcher will omitting `make` or `$(MAKE)` work with other non-make commands. The whole `deploy` command looks like 
```
 deploy:
 	make build-webpack
  	node ./build/deploy/s3.js
  	$(MAKE) build-css-loader
  	$(MAKE) npm-publish
```

File: Makefile (hunk 2 above)
Comment (leggsimon) [inline Makefile]: Actually, I think I'd rather it just be part of the deploy step. I'd be nervous about doing the `build-production` and then `test` incase `test` generates stuff we don't want to be deployed.

File: Makefile (hunk 1 above)
Comment (leggsimon) [inline Makefile]: So TIL, `$(MAKE)` creates a new instance of make and then runs that command. we don't need to do that

File: Makefile (hunk 2 above)
Comment (bjfletcher) [inline Makefile]: @leggsimon `test` task should absolutely not generate anything for deployment... it should only test
I think all developers are clear about that
Anything generated for deployment should be in the `build` step, never `test`.
If you see any project that generates/builds in the `test` step then that project should really move it into `build`.

Comment (leggsimon) [inline Makefile]: Have updated

File: Makefile [hunk 3]
```diff
@@ -61,6 +64,17 @@ copy-stylesheet-partial:
...
+build-bundle:
+	webpack --bail --config build/deploy/webpack.config.js
+
+deploy-s3:
+	node ./build/deploy/s3.js
+
+rebuild-user-facing-apps:
+	# only autodeploy all apps in office hours
+	HOUR=$$(date +%H); DAY=$$(date +%u); if [ $$HOUR -ge 8 ] && [ $$HOUR -lt 16 ] && [ $$DAY -ge 0 ] && [ $$DAY -lt 6 ]; then \
+	echo "REBUILDING ALL APPS" && sleep 20 && nht rebuild --all --serves user-page; fi
```
Comment (bjfletcher) [inline Makefile]: "In 20 seconds we will start rebuilding ALL apps... quit now if you don't want to!"? :)

File: components/n-ui/tracking/third-party/chartbeat.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ Eventually this will be replaced by FT's in-house "lantern" application, utilisi
 */
 
 import {cookieStore} from 'n-ui-foundations';
-import {loadScript} from '../../../../browser/bootstrap/js/utils';
+import {loadScript} from '../../../../browser/js/utils';
```
Comment (bjfletcher) [inline components/n-ui/tracking/third-party/chartbeat.js]: a separate bug?

Comment (leggsimon) [inline components/n-ui/tracking/third-party/chartbeat.js]: Yep, this is actually already in master but for some reason has been added in this PR. A rebase gone awry I think.

File: Makefile (hunk 3 above)
Comment (leggsimon) [inline Makefile]: https://github.com/Financial-Times/n-ui/issues/972

**Pull request merged**